src/include/txn.i

```diff
@@ -797,13 +797,12 @@ __wt_txn_read_upd_list(WT_SESSION_IMPL *session, WT_CURSOR_BTREE *cbt, WT_UPDATE
             continue;
         upd_visible = __wt_txn_upd_visible_type(session, upd);
         if (upd_visible == WT_VISIBLE_TRUE) {
-            /*
-             * A tombstone representing a stop time pair will have either a valid txn id or a valid
-             * timestamp. Ignore such tombstones in history store based on session settings.
-             */
-            if (type == WT_UPDATE_TOMBSTONE && WT_IS_HS(S2BT(session)) &&
-              F_ISSET(session, WT_SESSION_IGNORE_HS_TOMBSTONE) &&
-              (upd->start_ts != WT_TS_NONE || upd->txnid != WT_TXN_NONE))
+            /* Ignore non-globally visible tombstones when we are doing history store scans in
+             * rollback to stable or when we are told to. */
```

Comment:
The comment here is wrong. We should add a check in s_all that catches this style of comment - I've noticed a few in reviews lately.